azurerm/resource_arm_recovery_services_fabric.go

```diff
@@ -0,0 +1,147 @@
+			},
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"recovery_vault_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: azure.ValidateRecoveryServicesVaultName,
+			},
+			"location": azure.SchemaLocation(),
+		},
+	}
+}
+
+func resourceArmRecoveryServicesFabricCreate(d *schema.ResourceData, meta interface{}) error {
+	resGroup := d.Get("resource_group_name").(string)
+	vaultName := d.Get("recovery_vault_name").(string)
+	location := azure.NormalizeLocation(d.Get("location").(string))
+	name := d.Get("name").(string)
+
+	client := meta.(*ArmClient).recoveryServices.FabricClient(resGroup, vaultName)
+	ctx := meta.(*ArmClient).StopContext
+
+	if requireResourcesToBeImported && d.IsNewResource() {
+		existing, err := client.Get(ctx, name)
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("Error checking for presence of existing rec	overy services fabric %s (vault %s): %+v", name, vaultName, err)
+			}
+		}
+
+		if existing.ID != nil && *existing.ID != "" {
+			return tf.ImportAsExistsError("azurerm_resource_group", azure.HandleAzureSdkForGoBug2824(*existing.ID))
+		}
+	}
+
+	parameters := siterecovery.FabricCreationInput{
+		Properties: &siterecovery.FabricCreationInputProperties{
+			CustomDetails: siterecovery.AzureFabricCreationInput{
+				InstanceType: "Azure",
+				Location:     &location,
+			},
+		},
+	}
+
+	future, err := client.Create(ctx, name, parameters)
+	if err != nil {
+		return fmt.Errorf("Error creating recovery services fabric %s (vault %s): %+v", name, vaultName, err)
+	}
+	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("Error creating recovery services fabric %s (vault %s): %+v", name, vaultName, err)
+	}
+
+	resp, err := client.Get(ctx, name)
+	if err != nil {
+		return fmt.Errorf("Error retrieving recovery services fabric %s (vault %s): %+v", name, vaultName, err)
+	}
+
+	d.SetId(azure.HandleAzureSdkForGoBug2824(*resp.ID))
+
+	return resourceArmRecoveryServicesFabricRead(d, meta)
+}
+
+func resourceArmRecoveryServicesFabricRead(d *schema.ResourceData, meta interface{}) error {
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resGroup := id.ResourceGroup
+	vaultName := id.Path["vaults"]
+	name := id.Path["replicationFabrics"]
+
+	client := meta.(*ArmClient).recoveryServices.FabricClient(resGroup, vaultName)
+	ctx := meta.(*ArmClient).StopContext
+
+	resp, err := client.Get(ctx, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error making Read request on recovery services fabric %s (vault %s): %+v", name, vaultName, err)
+	}
+
+	d.Set("name", resp.Name)
+	d.Set("resource_group_name", resGroup)
+	if props := resp.Properties; props != nil {
+		d.Set("location", props.FriendlyName) // Crazy? yes. But the location comes back in the friendly name
```

Comment:
It might make sense to open an azure sdk bug for this one? 🤔

Reply:
I found the correct way to get the location now that I have worked a bit more with the API.